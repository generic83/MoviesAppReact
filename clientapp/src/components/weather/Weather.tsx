import React, { useState } from "react";
import WeatherSummary from "./WeatherSummary";
import WeatherList from "./WeatherList";

const Weather = () => {
  const [message, setMessage] = useState("Default message");
  const changeMessage = (text: string) => {
    setMessage(text);
  };
  return (
    <React.Fragment>
      <WeatherSummary message={message} changeMessage={changeMessage} />
      <WeatherList />
    </React.Fragment>
  );
};

export default Weather;
