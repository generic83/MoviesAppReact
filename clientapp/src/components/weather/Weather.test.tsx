import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Weather from "./Weather";

//To update snapshot: npm test -- --updateSnapshot
//https://stackoverflow.com/questions/50138532/command-not-found-jest
//https://jestjs.io/docs/snapshot-testing

describe("Weather component", () => {
  test("should contain 'Default message'", () => {
    render(<Weather />);
    const element = screen.getByText(/Default message/i);
    expect(element).toBeInTheDocument();
  });

  test("should change default message", () => {
    //Arrange
    render(<Weather />);

    //Act
    const btn = screen.getByText("Change message"); //queryByText returns null in case no element found
    userEvent.click(btn);

    const element = screen.getByText(/This is a message with/i);
    expect(element).toBeInTheDocument();
  });

  test("test snapshot", () => {
    const { container, getByText } = render(<Weather />);
    //more about snapshots here
    expect(getByText(/Default message/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
