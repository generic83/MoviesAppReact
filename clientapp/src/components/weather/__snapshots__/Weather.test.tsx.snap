// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component test snapshot 1`] = `
<div>
  <h2>
    Default message
  </h2>
  <button>
    Change message
  </button>
  <div
    class="grid"
  >
    <table>
      <tbody />
    </table>
    <button>
      Refresh
    </button>
  </div>
</div>
`;
